lib/mongo/collection.rb

```diff
@@ -692,7 +692,7 @@ def insert_one(document, opts = {})
     # @example Insert documents into the collection.
     #   collection.insert_many([{ name: 'test' }])
     #
-    # @param [ Array<Hash> ] documents The documents to insert.
+    # @param [ Array<Hash>, Enumerable ] documents The documents to insert.
```

Comment:
I don't use type in Ruby, so I don't know which one can be generic.
However, if Enumerable can be generic too and `Array` is also an `Enumerable`, we can also just do `Enumerable<Hash>`.  

There are three possibilities forward:
* `Array<Hash>, Enumerable` as Neil suggested  
probably maybe because `Enumerable` is not a generic type? And also not to confuse developers that are new to the concept of Ruby's module.
* `Array<Hash>, Enumerable<Hash>` as you suggested  
because `Enumerable` is actually a generic type; leaving the `Array` type as is for the same reason as above.
* `Enumerable<Hash>`  
if `Enumerable` is a generic type, and we prioritize conciseness.

Reply:
I don't have a strong opinion on this, I had just looked for examples elsewhere in the code base and this is what we did. I don't mind switching it...

Reply:
I think `Array<Hash> | Enumerable<Hash>` is redundant, my vote is for `Enumerable<Hash>` only.

@midnight-wonderer This notation simply signifies the expected type of values in the container, we haven't magically come up with a way to add generics to Ruby :)

Reply:
That's fine with me

Reply:
done